ansible-tests/validations/library/validate_network_environment.py

```diff
@@ -135,6 +139,56 @@ def check_allocation_pools_pairing(filedata, pools):
     return errors
 
 
+def check_static_ip_pool_collision(static_ips, pools):
+    """Statically defined IP address must not conflict with allocation pools.
+
+    The allocation pools come as a dict of items in the following format:
+
+    InternalApiAllocationPools: [{'start': '10.35.191.150', 'end': '10.35.191.240'}]
+
+    The static IP addresses are dicts of:
+
+    ComputeIPs: {
+        'internal_api': ['10.35.191.100', etc.],
+        'storage': ['192.168.100.45', etc.]
+    }
+    """
+    errors = []
+
+    pool_ranges = []
+    for pool_name, ranges in six.iteritems(pools):
+        for allocation_range in ranges:
+            ip_range = netaddr.IPRange(allocation_range['start'],
+                                allocation_range['end'])
+            pool_ranges.append((pool_name, ip_range))
+    for role, services in six.iteritems(static_ips):
+        for service, ips in six.iteritems(services):
+            for ip in ips:
+                ranges_with_conflict = ip_conflicts_with_ranges(ip, pool_ranges)
+                if ranges_with_conflict:
+                    for pool_name, ip_range in ranges_with_conflict:
+                        msg = "IP address {} from {}[{}] is in the {} pool."
+                    errors.append(msg.format(
+                        ip, role, service, pool_name))
+    return errors
+
+
+def ip_conflicts_with_ranges(ip_address, ip_ranges):
```

Comment:
It's not clear from the name of the function to guess it filters ranges to the ones conflicting with the given IP address.

I suggest naming it backward, `ranges_conflicting_with_ip` or similar to make it clear it returns ranges.